controllers/upgrade/upgradeExecutor_test.go

```diff
@@ -159,14 +161,24 @@ func Test_upgradeExecutor_Upgrade(t *testing.T) {
 			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Extracting optional custom K8s resources...").Once().
 			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Applying/Updating custom dogu resources to the cluster: [%s]", "my-custom-resource.yml").Once().
 			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Updating dogu resources in the cluster...").Once().
+			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Successfully updated health status to %q", k8sv1.UnavailableHealthStatus).Once().
 			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Applying optional post-upgrade scripts...").Once().
 			On("Eventf", toDoguResource, typeNormal, upgradeEvent, "Reverting to original startup probe values...").Once()
 
 		execPodFactory := mocks.NewExecPodFactory(t)
 		execPodFactory.On("NewExecPod", toDoguResource, toDogu).Return(execPod, nil)
 
+		ecosystemClientMock := mocks.NewEcosystemInterface(t)
+		doguClientMock := mocks.NewDoguInterface(t)
+		ecosystemClientMock.EXPECT().Dogus("").Return(doguClientMock)
+		doguClientMock.EXPECT().UpdateStatusWithRetry(testCtx, toDoguResource, mock.Anything, mock.Anything).RunAndReturn(func(ctx context.Context, dogu *k8sv1.Dogu, f func(k8sv1.DoguStatus) k8sv1.DoguStatus, options metav1.UpdateOptions) (*k8sv1.Dogu, error) {
+			toDoguResource.Status.Health = k8sv1.UnavailableHealthStatus
```

Comment:
You expect that your test code sets the status correctly.
Call the function `f` here with the status of `toDoguResource` as parameter.